zulipterminal/cli/run.py

```diff
@@ -206,31 +207,46 @@ def styled_input(label: str) -> str:
     return input(in_color("blue", label))
 
 
-def get_login_id(server_properties: Dict[str, Any]) -> str:
+def get_login_label(server_properties: ServerSettings) -> str:
     require_email_format_usernames = server_properties["require_email_format_usernames"]
     email_auth_enabled = server_properties["email_auth_enabled"]
 
     if not require_email_format_usernames and email_auth_enabled:
-        label = "Email or Username: "
+        return "Email or Username: "
     elif not require_email_format_usernames:
-        label = "Username: "
+        return "Username: "
     else:
         # TODO: Validate Email address
-        label = "Email: "
+        return "Email: "
 
-    return styled_input(label)
+
+class NotAZulipOrganizationError(Exception):
+    pass
+
+
+def get_server_settings(realm_url: str) -> ServerSettings:
+    response = requests.get(url=f"{realm_url}/api/v1/server_settings")
+    unsuccessful_response = response.status_code != requests.codes.OK
+    response_realm_name = response.json().get("realm_name")
```

Comment:
Maybe only call this line if you have a successful response?  I can imagine a 500 error code with a non-JSON body, which would cause this line to throw an exception.

Reply:
Agreed. Reading this again, I suspect this last commit was a less thought-through exploration of that discussion, rather than the earlier tidying and refactoring.

Probably good to get the earlier commits in soon after the topic on czo, but maybe just dropping that line and replacing the conditional below with:
```python3
if unsuccessful_response or "realm_name" not in response.json():
    ...
```
(plus tests)